coderd/notifications/dispatch/smtp.go

```diff
@@ -0,0 +1,338 @@
+package dispatch
+
+import (
+	"bytes"
+	"context"
+	_ "embed"
+	"fmt"
+	"mime/multipart"
+	"mime/quotedprintable"
+	"net"
+	"net/mail"
+	"net/smtp"
+	"net/textproto"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	"golang.org/x/xerrors"
+
+	"cdr.dev/slog"
+	"github.com/coder/coder/v2/coderd/database"
+	"github.com/coder/coder/v2/coderd/notifications/render"
+	"github.com/coder/coder/v2/coderd/notifications/types"
+	"github.com/coder/coder/v2/codersdk"
+)
+
+var (
+	ValidationNoFromAddressErr   = xerrors.New("no 'from' address defined")
+	ValidationNoSmarthostHostErr = xerrors.New("smarthost 'host' is not defined, or is invalid")
+	ValidationNoSmarthostPortErr = xerrors.New("smarthost 'port' is not defined, or is invalid")
+	ValidationNoHelloErr         = xerrors.New("'hello' not defined")
+
+	//go:embed smtp/html.gotmpl
+	htmlTemplate string
+	//go:embed smtp/plaintext.gotmpl
+	plainTemplate string
+)
+
+// SMTPHandler is responsible for dispatching notification messages via SMTP.
+// NOTE: auth and TLS is currently *not* enabled in this initial thin slice.
+// TODO: implement auth
+// TODO: implement TLS
```

Comment:
link follow-up issue for this TODO

Reply:
https://github.com/orgs/coder/projects/15/views/26?pane=issue&itemId=69135708